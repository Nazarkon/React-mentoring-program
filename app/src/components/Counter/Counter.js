import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterNumber = (props) => {
  const { defaultValue } = props;

  const [number, setCount] = useState(defaultValue || 0);

  const incrementValue = () => {
    setCount((prevCount) => prevCount + 1);
  };

  const decrementValue = () => {
    setCount((prevCount) => prevCount - 1);
  };

  const IncrementButtonStyle = {
    backgroundColor: '#4CAF50',
    border: 'none',
    color: 'white',
    padding: '15px 32px',
    textAlign: 'center',
    textDecoration: 'none',
    display: 'inline-block',
    fontSize: '16px'
  };

  const DecrementButtonStyle = {
    backgroundColor: '#f44336',
    border: 'none',
    color: 'white',
    padding: '15px 32px',
    marginLeft: '10px',
    textAlign: 'center',
    textDecoration: 'none',
    display: 'inline-block',
    fontSize: '16px'
  };

  return React.createElement(
    'div',
    null,
    React.createElement('h1', null, `Count: ${number}`),
    React.createElement(
      'button',
      { style: IncrementButtonStyle, onClick: incrementValue },
      'Increment'
    ),
    React.createElement(
      'button',
      { style: DecrementButtonStyle, onClick: decrementValue },
      'Decrement'
    )
  );
};

CounterNumber.propTypes = {
  defaultValue: PropTypes.number
};

export default CounterNumber;
