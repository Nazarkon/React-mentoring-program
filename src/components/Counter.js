import React, {useState} from "react";


const CounterNumber = (props) => {

    const { defaultValue } = props

    const [number, setCount] = useState(defaultValue || 0);

    const incrementValue = () => {
        setCount(number + 1)
    }

    const decrementValue = () => {
        setCount(number - 1);
    }

    const IncrementButtonStyle = {
        backgroundColor: "#4CAF50",
        border: "none",
        color: "white",
        padding: "15px 32px",
        textAlign: "center",
        textDecoration: "none",
        display: "inline-block",
        fontSize: "16px",
    }

    const DecrementButtonStyle = {
        backgroundColor: "#f44336",
        border: "none",
        color: "white",
        padding: "15px 32px",
        marginLeft: "10px",
        textAlign: "center",
        textDecoration: "none",
        display: "inline-block",
        fontSize: "16px",
    }

    return React.createElement('div', null,
    React.createElement('h1', null, `Count: ${number}`),
    React.createElement('button', { style: IncrementButtonStyle, onClick: incrementValue }, 'Increment'),
    React.createElement('button', { style: DecrementButtonStyle , onClick: decrementValue }, 'Decrement')
  );
}

export default CounterNumber