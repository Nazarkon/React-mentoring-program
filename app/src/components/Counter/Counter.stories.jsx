import React from 'react';

import Counter from '../../components/Counter/Counter';

export default {
  title: 'Components/Counter',
  component: Counter,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    defaultValue: {
      control: 'number',
      description: 'Default counter value',
      defaultValue: 2
    }
  }
};

const Template = (args) => <Counter {...args} />;

export const Default = Template.bind({});
