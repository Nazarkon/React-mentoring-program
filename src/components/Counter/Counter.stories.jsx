import React from 'react';

import Counter from './Counter';

export default {
  title: 'Components/Counter',
  component: Counter,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    defaultValue: {
      control: 'number',
      description: 'Default counter value',
      defaultValue: 2,
    },
  },
};

function Template(args) {
  return <Counter {...args} />;
}

export const Default = Template.bind({});
