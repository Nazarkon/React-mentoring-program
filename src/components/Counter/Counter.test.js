import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import Counter from "./Counter";
import { act } from "react-dom/test-utils";

describe("Check Counter element behavior", () => {
  it("Matches snapshot", () => {
    const { container } = render(<Counter />);
    expect(container).toMatchSnapshot();
  });

  it("should display the default count value", () => {
    const { getByText } = render(<Counter defaultValue={0} />);
    expect(getByText("Count: 0")).toBeInTheDocument();
  });

  it("should increment the count value when the increment button is clicked", () => {
    const { getByText } = render(<Counter defaultValue={0} />);
    const incrementButton = getByText("Increment");
    act(() => {
      userEvent.click(incrementButton);
    });
    expect(getByText("Count: 1")).toBeInTheDocument();
  });

  it("should decrement the count value when the decrement button is clicked", () => {
    const { getByText } = render(<Counter defaultValue={0} />);
    const decrementButton = getByText("Decrement");
    act(() => {
      userEvent.click(decrementButton);
    });
    expect(getByText("Count: -1")).toBeInTheDocument();
  });
});
