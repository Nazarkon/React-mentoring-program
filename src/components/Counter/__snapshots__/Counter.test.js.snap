// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Counter element behavior Matches snapshot 1`] = `
<div>
  <div>
    <h1>
      Count: 0
    </h1>
    <button
      style="background-color: rgb(76, 175, 80); color: white; padding: 15px 32px; text-align: center; text-decoration: none; display: inline-block; font-size: 16px;"
    >
      Increment
    </button>
    <button
      style="background-color: rgb(244, 67, 54); color: white; padding: 15px 32px; margin-left: 10px; text-align: center; text-decoration: none; display: inline-block; font-size: 16px;"
    >
      Decrement
    </button>
  </div>
</div>
`;
